Mailgun notify service

pschmitt opened the pull request:
**Description:**
This implements a new notification service: [Mailgun](https://mailgun.com). It is somewhat similar to the SMTP notifier (it sends mail) but instead of SMTP it uses [Mailgun's REST API](https://documentation.mailgun.com/api-sending.html).

**Pull request in [home-assistant.github.io](https://github.com/home-assistant/home-assistant.github.io) with documentation (if applicable):** home-assistant/home-assistant.github.io#1989

**Example entry for `configuration.yaml` (if applicable):**
```yaml
notify:
  - platform: mailgun
    domain: mg.example.com
    sandbox: False
    sender: hass@mg.example.com
    recipient: me@example.com
    token: 'key-XXXXXXXXXXXXX'

```

**Checklist:**

If user exposed functionality or configuration variables are added/changed:
  - [x] Documentation added/updated in [home-assistant.github.io](https://github.com/home-assistant/home-assistant.github.io)

If the code communicates with devices, web services, or third-party tools:
  - [x] Local tests with `tox` run successfully. **Your PR cannot be merged unless tests pass**
  - [x] New dependencies have been added to the `REQUIREMENTS` variable ([example][ex-requir]).
  - [x] New dependencies are only imported inside functions that use them ([example][ex-import]).
  - [x] New dependencies have been added to `requirements_all.txt` by running `script/gen_requirements_all.py`.
  - [x] New files were added to `.coveragerc`.

If the code does not interact with devices:
  - [ ] Local tests with `tox` run successfully. **Your PR cannot be merged unless tests pass**
  - [ ] Tests have been added to verify that the new code works.

[ex-requir]: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard.py#L16
[ex-import]: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard.py#L51

pschmitt committed:
Mailgun notify service

balloob reviewed (commented):
Cool!

Would you be able to verify the credentials before returning the service?

pschmitt committed:
Update dependency to version 1.3

- The provided credentials (including the domain) are now checked during
startup, as requested by @balloob
- The domain name is now optional
- There's a new config item "sandbox" which indicates whether to use the
sandboxed domain in case the domain is not set

pschmitt committed:
Fix a few lint issues

pschmitt committed:
Disable lint check no-value-for-parameter

balloob commented:
Nice!
